pkg/vali/validation.go

```diff
@@ -0,0 +1,99 @@
+package vali
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+var mailRe = regexp.MustCompile(`\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z`)
+
+const tagName = "validate"
+
+type Validator interface {
```

Comment:
My personal opinion is that I would name this package validation and this interface as `Interface`.